shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Rewards.hs

```diff
@@ -373,13 +373,14 @@ rewardOnePool pp r blocksN blocksTotal pool (Stake stake) sigma sigmaA (Coin tot
     appPerf = mkApparentPerformance (_d pp) sigmaA blocksN blocksTotal
     poolR = rationalToCoinViaFloor (appPerf * fromIntegral maxP)
     tot = fromIntegral total
+    poolHashes = (KeyHashObj `Set.map` _poolOwners pool)
     mRewards =
       Map.fromList
         [ ( hk,
             memberRew poolR pool (StakeShare (fromIntegral c % tot)) (StakeShare sigma)
           )
           | (hk, Coin c) <- Map.toList stake,
-            hk `Set.notMember` (KeyHashObj `Set.map` _poolOwners pool)
+            hk `Set.notMember` poolHashes
```

Comment:
Can't we avoid this O(n) operation entirely:
```
poolHashes = (KeyHashObj `Set.map` _poolOwners pool)
```
We just need to check if `kh` is in `poolOwners pool`, something like
```
notPoolOwner (KeyHashObj hk) = hk `Set.notMember` _poolOwners pool
notPoolOwner (ScriptHashObj hk) = False
```
right?

Reply:
yep, `notPoolOwner` is all we need